lib/Doctrine/DBAL/Driver/OCI8/OCI8Statement.php

```diff
@@ -361,63 +362,63 @@ public function getIterator()
     /**
      * {@inheritdoc}
      */
-    public function fetch($fetchMode = null, ...$args)
+    public function fetch(?FetchMode $fetchMode = null, ...$args)
     {
         // do not try fetching from the statement if it's not expected to contain result
         // in order to prevent exceptional situation
         if (!$this->result) {
             return false;
         }
 
-        $fetchMode = $fetchMode ?: $this->_defaultFetchMode;
+        $fetchMode = $fetchMode ?? $this->_defaultFetchMode;
 
-        if ($fetchMode === FetchMode::COLUMN) {
+        if ($fetchMode === FetchMode::COLUMN()) {
             return $this->fetchColumn();
         }
 
-        if ($fetchMode === FetchMode::STANDARD_OBJECT) {
+        if ($fetchMode === FetchMode::STANDARD_OBJECT()) {
             return oci_fetch_object($this->_sth);
         }
 
-        if (! isset(self::$fetchModeMap[$fetchMode])) {
-            throw new \InvalidArgumentException("Invalid fetch style: " . $fetchMode);
+        if (! isset(self::$fetchModeMap[$fetchMode()])) {
```

Comment:
I don't think this will work as expected

Reply:
I think it will - `()` is an accessor for the value the enum holds. And since PHP arrays don't allow maps from objects to mixed, it's for convenience - yes, it could be SplObjectStorage<SomeEnum, mixed> as well.